### PR TITLE
API to rename licence and create event for KPI data

### PR DESCRIPTION
**Pull request by Stephan-Hough**
feat/water-2644
-- postLicenceName route added to rename licence
-- controller handler post method added
-- crm documents method to update documentHeader
-- tests added and updated

### CODE REVIEW

File: src/modules/licences/controller.js [hunk 1]
```diff
@@ -292,10 +292,27 @@ const getLicenceCompanyByDocumentId = async (request, h) => {
   }
 };
 
+const postLicenceName = async (request, h) => {
+  const { documentId } = request.params;
+  const { documentName, rename } = request.payload;
```
Comment (james-siteclick) [inline src/modules/licences/controller.js:297]: I think the value of `rename` can be established without needing an extra payload parameter, along the lines of:
```
const { isEmpty } = require('lodash');
// ...
const { data : { document_name : currentName } } = await documentsClient.findOne(documentId);
const rename = !isEmpty(currentName);
```

File: src/modules/licences/routes.js [hunk 2]
```diff
@@ -126,5 +126,24 @@ module.exports = {
...
+    handler: controller.postLicenceName,
+    config: {
+      description: 'Sets name for a licence',
+      validate: {
+        params: {
+          documentId: Joi.string().guid().required()
+        },
+        payload: {
+          documentName: Joi.string().required().allow(''),
+          rename: Joi.boolean().required(),
```
Comment (james-siteclick) [inline src/modules/licences/routes.js:143]: As above, not sure the rename param is needed

File: src/modules/licences/controller.js (hunk 1 above)
Comment (Stephan-Hough) [reply src/modules/licences/controller.js]: As discussed in the SLACK the logic should not sit with the UI so this has been implemented as requested

File: src/modules/licences/routes.js (hunk 2 above)
Comment (Stephan-Hough) [reply src/modules/licences/routes.js]: done

**Pull request merged**